estatioapp/dom/src/main/java/org/estatio/dom/party/Organisation.java

```diff
@@ -82,28 +90,57 @@ public ApplicationTenancy getApplicationTenancy() {
 
     // //////////////////////////////////////
 
-    @Property(optionality = Optionality.OPTIONAL)
+    @Persistent(mappedBy = "organisation")
+    @CollectionLayout(defaultView = "table")
+    @Getter @Setter
+    private SortedSet<OrganisationPreviousName> previousNames = new TreeSet<>();
+
+    // //////////////////////////////////////
+
     public Organisation change(
-            final String name,
             final @Parameter(optionality = Optionality.OPTIONAL, regexPattern = RegexValidation.REFERENCE, regexPatternReplacement = RegexValidation.REFERENCE_DESCRIPTION) String vatCode,
             final @Parameter(optionality = Optionality.OPTIONAL, regexPattern = RegexValidation.REFERENCE, regexPatternReplacement = RegexValidation.REFERENCE_DESCRIPTION) String fiscalCode) {
-        setName(name);
         setVatCode(vatCode);
         setFiscalCode(fiscalCode);
 
         return this;
     }
 
     public String default0Change() {
-        return getName();
+        return getVatCode();
     }
 
     public String default1Change() {
-        return getVatCode();
+        return getFiscalCode();
     }
 
-    public String default2Change() {
-        return getFiscalCode();
+    public Organisation changeName(
+            final String name,
+            final LocalDate previousNameEndDate) {
+        if (!name.equals(getName())) {
+            OrganisationPreviousName organisationPreviousName = organisationPreviousNameRepository.newOrganisationPreviousName(getName(), previousNameEndDate);
+            getPreviousNames().add(organisationPreviousName);
+        }
+
+        setName(name);
+
+        return this;
+    }
+
+    public String default0ChangeName() {
+        return getName();
+    }
+
+    public LocalDate default1ChangeName() {
+        return getClockService().now();
+    }
+
+    public String validateChangeName(
```

Comment:
Perhaps there should be some validation that if the name is the same as the current name, then to reject (ie the new name must be different from the current name).